### PR TITLE
Contexts with elements

### PR DESCRIPTION
**Pull request by benjay10**
This pull request introduces the features `ctx.spawn.**` and overrideUri, where `**` are selected HTML element tags such as `span`, `div`, `li`, `a` and even nested ones like `lia`. Just like `<WithRdfaContext>`, `<ctx.spawn.**>` will create an HTML tag with the RDFa properties added, instead of having to use the repetitive construct with the `{{rdfa}}` modifier.

```handlebars
<ctx.get @prop="bestuursorgaan" as |elements value boCtx|>
  <span {{rdfa elements}}>
    <boCtx.get @prop="isTijdsspecialisatieVan" as |elements value itsvCtx|>
      <span {{rdfa elements}}>
        <itsvCtx.get @prop="naam" />
      </span>
    </boCtx.get>
  </span>
</ctx.get>
```

could be replaced by this:

```handlebars
<ctx.span @prop="bestuursorgaan" as |value boCtx|>
  <boCtx.span @prop="isTijdsspecialisatieVan" as |value itsvCtx|>
    <itsvCtx.get @prop="naam" />
  </boCtx.span>
</ctx.span>
```

Maybe not the best for performance, as we are now creating more nested components, but absolutely necessary to work with FastBoot, where modifiers like `{{rdfa}}` are not run.

Ah, and there were also some bugs from master that were fixed during this PR.

EDIT: the examples above reflect the most recent implementation now.

### CODE REVIEW

**Comment by benjay10** [comment]
I'm changing the implementation according to direction from @madnificent. They are simpler, and still have the same power.

There is one drawback, as far as I can see at the moment: because the use of the term `spawn` is too verbose, the component nesting should be flattened, and this makes `ctx.div` available next to `ctx.get`. However, iteration with 'each' also needs to support multiple sub-components, and this is why `ctx.each` does not exist anymore as it represents the hash map for deeper nesting of components.

Repositories using `ctx.each` should use `ctx.each.get` instead.

File: tests/integration/components/rdfa/ctx/span-test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { module, test } from 'qunit';
```
Comment (nvdk) [inline tests/integration/components/rdfa/ctx/span-test.js:1]: I'm typically not a fan of keeping the standard tests, they add little value

**Review by nvdk — APPROVED** [review]
Looks good and from basic testing seems to work ok.  I'm not super familiar with this addon so I don't have strong opinions on how this was implemented, perhaps best to see if @Windvis or @madnificent have a more informed look.

**Comment by benjay10** [comment]
> Looks good and from basic testing seems to work ok. I'm not super familiar with this addon so I don't have strong opinions on how this was implemented, perhaps best to see if @Windvis or @madnificent have a more informed look.

Removed them. I always try to remove empty tests, but I forgot them this time.

**Comment by Windvis** [comment]
I don't really have an opinion about the new components since I'm not very familiar with adding RDFa attributes. I think this is one for @madnificent to review :smile:.

I would add tests for the new components though. Probably as part of ctx-get and ctx-each makes the most sense since that's the public API? I think verifying that the correct tags are rendered and the correct attributes are set makes sense and shouldn't be that hard?

Also, the ctx-each tests are failing at the moment :smile: Drone is still configured that a failing "tests" step is ignored.

**Comment by benjay10** [comment]
Sorry, added bug label back, because there is at least one bug fixed with this PR: https://github.com/lblod/ember-rdfa-helpers/pull/12/commits/b5b14b3bf051e5148943c42ab76168c0c699b2d9

**Comment by Windvis** [comment]
I might be wrong but I think you should just use the label that makes the most sense, which is 'breaking' for this PR. 

I'm not sure if lerna changelog is smart enough to "deduplicate" things, but if not it seems weird to show the same PR title 3 times in the changelog under different sections. So I think only "breaking" is needed here.

**Comment by nvdk** [comment]
agree with @Windvis

**Comment by benjay10** [comment]
I introduced the `.a` for when you want a block content in a @link that references another RDF individual. I think it used to just create a context, and not print any RDFa elements, as this was left to be done by the modifier (which doesn't run on FastBoot).

Subcomponents like `.p` and `.lia` where only introduced as a drop-in solution for some situation on the publication frontend, where I wanted to use these helpers to their fullest. I realise they can be replaced with some block content and a nested `.span`, but it's nice to see how easy it is to introduce a new subcomponent. Also, I think that with this set of subcomponents, most real world scenario's are covered. (?)

I will try to provide an example on `overrideUri`, but to be honest, the documentation on `useUri` is equally as vague to me as it also lacks an example. Maybe an example explaining both is needed.

**Comment by benjay10** [comment]
To quickly explain why I thought `.a` was a good idea, consider the following identical (in result) blocks:

```handlebars
<ctx.get @prop="currentProject" @link={{true}} as |attributes value projCtx|>
  <a {{rdfa attributes}}>
    <projCtx.get @prop="name" />
  </a>
</ctx.get>

<ctx.a @prop="currentProject" @link={{true}} as |value projCtx|>
  <projCtx.get @prop="name" />
</ctx.a>
```

If you cannot run the modifier (e.g. FastBoot), the second block with the `.a` seemed like the only clean solution. I realise that if you use `.p` instead, that it would add `href` to a `<p>` and that makes no sense.

I hope this at least lets you understand why it was done this way. I'm currently writing tests for this. They are also a good source of examples.

**Pull request merged**